ui/app/components/auth-jwt.js

```diff
@@ -109,7 +109,11 @@ export default Component.extend({
   },
 
   exchangeOIDC: task(function*(event, oidcWindow) {
-    let oidcState = event.storageArea.getItem('oidcState');
+    // in non-incognito mode we need to use a timeout because it takes time before oidcState is written to local storage.
+    let oidcState = Ember.testing
+      ? event.storageArea.getItem('oidcState')
+      : yield timeout(1000).then(() => event.storageArea.getItem('oidcState'));
```

Comment:
I'm not sure if this works on `localStorage`, but it could be worth trying `waitForProperty` to see if 'oidcState' is set instead of `waitForEvent` which seems to be failing in a strange way (the event being fired but the value isn't available). http://ember-concurrency.com/api/global.html#waitForProperty

Reply:
I missed this comment before I merged (I didn't refresh).  I'll play around with this and see if it's a better solution.  My apologies for not seeing this.

Reply:
It doesn't appear that the `watiForProperty` solves the issue.  Thank you for the suggestion!